### PR TITLE
Restores yes/no/on/off for booleans on the command line

### LINKED ISSUE #2732: conda config --set show_channel_urls yes doesn't work anymore
This is happening since the latest conda update:

```bat
λ conda config --set show_channel_urls yes
Error: Key: show_channel_urls; yes is not a YAML boolean.
```

It happens with both conda 4.1.1 (local windows py 3.5) and 4.1.0 (appveyor, https://ci.appveyor.com/project/mdboom/matplotlib/build/1.0.1774) and it worked with 4.0.8 (https://ci.appveyor.com/project/mdboom/matplotlib/build/1.0.1765/job/bkldg98f8p087xmf)

### PR DESCRIPTION
**Pull request by mcg1969**
Fixes #2732

### CODE REVIEW

**Comment by janschulz** [comment]
IMO it would be good to add a deprecation warning for the "wrong" values

**Comment by mcg1969** [comment]
Actually, @janschulz, I see no reason to deprecate them at all. It's clear from the updated `test_config.py` it was intentional to remove their support, but if people are using them, I just don't see why we shouldn't continue to do so. Booleans aren't hard to parse.

But here's a potentially important issue, @kalefranz, @msarahan. In `test_set_rc_string`, I see this comment:
```
    # We specifically test ssl_verify since it can be either a boolean or a string
``` 
Well, but that isn't really true, is it? After all, `ssl_verify` is in `rc_string_keys`, and the test was clearly changed to verify that `yes` is preserved as a string. So it's no longer true that `ssl_verify` is a boolean.

But that got me thinking: is _another_ part of the code expecting `ssl_verify` to be either a string or a boolean? The only place I see it is in `connection.py`, where it is used to initialize `self.verify` in the `CondaSession` object. The base `requests.Session` object uses this; see its docs [here](http://requests.readthedocs.io/en/master/api/?highlight=verify#requests.Session.verify); it _does_ expect it to be a boolean or a CA_BUNDLE path.

My point: I think `ssl_verify` _is_ intended to be interpreted as a YAML boolean whenever it is possible to do so. I've therefore modified the `CondaSession` constructor to use `yaml_bool` with a passthrough if it fails. Could this explain some of our stranger SSL errors?

**Comment by mcg1969** [comment]
Verified that the stricter YAML 1.2 convention was intentional: https://github.com/conda/conda/pull/2077

**Comment by mcg1969** [comment]
As a compromise, we could drop on/off and just keep yes/no.

**Comment by mcg1969** [comment]
Moved to WIP until we discuss. But note: this _only_ fixes the command-line parsing. It still forces strict adherence to 1.2 in the files themselves.

File: conda/utils.py [hunk 1]
```diff
@@ -292,6 +292,20 @@ def get_yaml():
     return yaml
 
 
+# Restores YAML 1.1 boolean flexibility.
+yaml_bool_ = {
+    'true': True, 'yes': True, 'on': True,
+    'false': False, 'no': False, 'off': False
+}
+def yaml_bool(s, passthrough=None):
+    if type(s) is bool:
```
Comment (kalefranz) [inline conda/utils.py]: I've always thought this was more semantic, but for whatever reason I've also always seen people say prefer `isinstance`.  Works for me for now though.

**Comment by kalefranz** [comment]
FWIW the 4.2 will probably use something like [boolify](https://github.com/kalefranz/auxlib/blob/develop/auxlib/type_coercion.py#L113).

File: conda/utils.py (hunk 1 above)
Comment (mcg1969) [inline conda/utils.py]: My view is that `isinstance` introduces unnecessary overhead when there is no possible need to deal with inheritance. I improved the performance of the solver code significantly by moving away from `isinstance`.

http://stackoverflow.com/questions/1549801/differences-between-isinstance-and-type-in-python
